### PR TITLE
[Ent-Corda]Adding idman chart

### PR DESCRIPTION
**Pull request by deepakkumardbd**
**Changelog**
- Add idman chart
 
**Reviewed by**
@jagpreetsinghsasan @suvajit-sarkar 

 

**Linked issue**
#777

### CODE REVIEW

File: platforms/corda-ent/charts/generate-pki/templates/job.yaml [hunk 1]
```diff
@@ -87,6 +87,7 @@ spec:
             KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
             VAULT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login -H "Content-Type: application/json" -d '{"role":"vault-role","jwt":"'"${KUBE_SA_TOKEN}"'"}' | jq -r 'if .errors then . else .auth.client_token end')
             validateVaultResponse 'vault login token' "${VAULT_TOKEN}"
+            VAULT_TOKEN=s.A8iSUYGGzCFypyoF5kKuoTLv
```
Comment (sownak) [inline platforms/corda-ent/charts/generate-pki/templates/job.yaml:90]: WHy is vault token checked in?

File: platforms/corda-ent/charts/idman/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  labels:
+    app.kubernetes.io/name: {{ .Values.nodeName }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ .Values.nodeName }}
```
Comment (sownak) [inline platforms/corda-ent/charts/idman/templates/deployment.yaml:20]: must match template.metadata.labels

File: platforms/corda-ent/charts/idman/templates/service.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  name: {{ .Values.nodeName }}
+  namespace: {{ .Values.metadata.namespace }}
+  labels:
+    run: {{ .Values.nodeName }}
+    app.kubernetes.io/name: {{ .Values.nodeName }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  selector:
```
Comment (sownak) [inline platforms/corda-ent/charts/idman/templates/service.yaml:13]: Ambassador annotations are missing.

File: platforms/corda-ent/charts/idman/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        volumeMounts:
+        - name: idman-logs
+          mountPath: /opt/corda/logs
+      imagePullSecrets:
+      - name: {{ .Values.dockerImage.imagePullSecret }}
+      volumes:
+        - name: idman-conf
+          configMap:
+            name: {{ .Values.nodeName }}-conf
+        - name: idman-etc
```
Comment (sownak) [inline platforms/corda-ent/charts/idman/templates/deployment.yaml:239]: etc is storing in memory and not PVC. Have you tested the idman pod comes up with exact same configurations when deleted?

File: platforms/corda-ent/charts/idman/templates/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  storageClassName: {{ .Values.storage.name }}
+  resources:
+    requests:
+      storage: 256Mi
+
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.nodeName }}-pvc-h2
```
Comment (sownak) [inline platforms/corda-ent/charts/idman/templates/pvc.yaml:28]: How will you ensure the PVCs do not get scheduled in a different AZ?

File: platforms/corda-ent/charts/idman/templates/deployment.yaml (hunk 4 above)
Comment (deepakkumardbd) [reply platforms/corda-ent/charts/idman/templates/deployment.yaml:239]: Yes I have tested it.

File: platforms/corda-ent/charts/idman/templates/pvc.yaml (hunk 5 above)
Comment (deepakkumardbd) [reply platforms/corda-ent/charts/idman/templates/pvc.yaml:28]: Changed volumeBindingMode in storageclass.tpl to ensure it.

File: platforms/corda-ent/charts/idman/templates/service.yaml (hunk 3 above)
Comment (deepakkumardbd) [reply platforms/corda-ent/charts/idman/templates/service.yaml:13]: Can't test it so not added for now.

**Pull request merged**